generators/delimited/main.py

```diff
@@ -0,0 +1,104 @@
+#!/usr/bin/env python3
+import sys
+import os
+import csv
+from time import strftime
+from unicode_csv import UnicodeDictReader
+
+# Minimum possible header.
+HEADER = (
+    'entity',
+    'attribute',
+    'value',
+)
+
+def generate(file_name, encoding, delimiter, identifier):
+    # Creating a CSV writer that feeds to stdout.
+    w = csv.writer(sys.stdout)
+
+    # Write the header to start.
+    w.writerow(HEADER)
+
+    # Create each fact
+    with open(file_name, 'rU') as f:
+        sniff = 1024
+        dialect = None
+
+        # Infer various properties about the file
+        # Sample the file to determine the dialect
+        sample = '\n'.join([l for l in f.readlines(sniff)])
+        f.seek(0)
+
+        # Determine dialect
+        sniffer = csv.Sniffer()
+        dialect = sniffer.sniff(sample)
+
+        reader = UnicodeDictReader(f, dialect=dialect, delimiter=delimiter, encoding=encoding)
+        if not identifier:
+            identifier = reader.fieldnames[0]
+
+        for line in reader:
+            # Assemble the identifier's fieldnames into a single string for output.
+            id = ''
+            if identifier is not None:
+                id = build_id(line, identifier)
+            else:
+                id = line[identifier]
```

Comment:
Now that `identifier` is guaranteed to not be empty, replace the above logic with just:

```python
id = build_id(line, identifier)
```